gas-tank/balance-services.md

```diff
@@ -0,0 +1,540 @@
+  })
+  .catch((error) => {
+    // Catch & handle the error
+  });
+```
+
+:::
+
+### Consume balance
+
+This endpoint allows users to consume their balance on the Gas Tank platform.
+
+To consume the balance, you need to sign the message with the private key of the address you want to consume the balance from. The `signature` field is the signature of the message signed by the address owner.
+
+Same as the [Increase balance](#increase-balance)/[Withdraw balance](#withdraw-balance), you can sign a string of data using [ethers.js](https://docs.ethers.org/v5/getting-started/), below is an example of [how to sign a message using ethers.js](https://docs.ethers.org/v5/getting-started/#getting-started--signing) and consume the balance.
+
+::: code-group
+
+```javascript [Request]
+import { ethers } from "ethers";
+
+const GAS_TANK_ENDPOINT = "https://gas-tank.xdefi.services";
+const message = "0x1234"; // Message to sign // [!code highlight]
+const privateKey = "0x1234"; // Private key of the address // [!code highlight]
+const wallet = new ethers.Wallet(privateKey);
+const signature = await wallet.signMessage(message);
+
+await fetch(`${GAS_TANK_ENDPOINT}/balances/consume`, {
+  method: "POST",
+  headers: {
+    "Content-Type": "application/json",
+    Authorization: `Bearer ${jwtToken}`, // JWT token // [!code highlight]
+  },
+  body: {
+    address: "string", // [!code highlight]
+    minDestinationAmount: "string", // [!code highlight]
+    destinationAddress: "string", // [!code highlight]
+    destinationChain: "string", // [!code highlight]
+    message: "string", // [!code highlight]
+    signature: "string", // [!code highlight]
+  },
+})
+  .then((response) => {
+    // Balance consumed successfully
+  })
+  .catch((error) => {
+    // Catch & handle the error
+  });
+```
+
+:::
+
+### Get a quote for consuming balance
+
+This endpoint allows users to generate a quote for consuming their balance on the Gas Tank platform.
+
+::: code-group
+
+```javascript [Request]
+const GAS_TANK_ENDPOINT = "https://gas-tank.xdefi.services";
+
+await fetch(`${GAS_TANK_ENDPOINT}/balances/consume/quote`, {
+  method: "POST",
+  headers: {
+    "Content-Type": "application/json",
+    Authorization: `Bearer ${jwtToken}`, // JWT token // [!code highlight]
+  },
+  body: {
+    address: "string", // [!code highlight]
+    minDestinationAmount: "string", // [!code highlight]
+    destinationChain: "string", // [!code highlight]
+  },
+})
+  .then((response) => {
+    console.log(response);
+    // Handle & do something with the response
+  })
+  .catch((error) => {
+    // Catch & handle the error
+  });
+```
+
+```json [Response]
+{
+  "expectedOutput": "string",
+  "fee": "string",
+  "spentAssets": [
+    {
+      "tokenAddress": "string",
+      "amount": "string",
+      "chain": "string"
+    }
+  ]
+}
+```
+
+:::
+
+### Get user balance queue update
+
+This endpoint allows users to get the balance queue update for a specific address.
```

Comment:
can you explain that it is to get the status of any job ongoing (deposit, consume, internal transfer, withdrawal)